llpc/translator/lib/SPIRV/SPIRVReader.cpp

```diff
@@ -4585,33 +4586,43 @@ Value *SPIRVToLLVM::transValueWithoutDecoration(SPIRVValue *bv, Function *f, Bas
   }
 
   case OpVectorShuffle: {
-    // NOTE: LLVM backend compiler does not well handle "shufflevector"
-    // instruction. So we avoid generating "shufflevector" and use the
-    // combination of "extractelement" and "insertelement" as a substitute.
     auto vs = static_cast<SPIRVVectorShuffle *>(bv);
 
     auto v1 = transValue(vs->getVector1(), f, bb);
     auto v2 = transValue(vs->getVector2(), f, bb);
 
-    auto vec1CompCount = vs->getVector1ComponentCount();
     auto newVecCompCount = vs->getComponents().size();
+    auto vec1CompCount = vs->getVector1ComponentCount();
+    auto vec2CompCount = vs->getVector2ComponentCount();
 
-    IntegerType *int32Ty = IntegerType::get(*m_context, 32);
-    Type *newVecTy = FixedVectorType::get(cast<VectorType>(v1->getType())->getElementType(), newVecCompCount);
-    Value *newVec = UndefValue::get(newVecTy);
+    llvm::SmallVector<int, 4> mask(vs->getComponents().begin(), vs->getComponents().end());
```

Comment:
You can do  `auto mask = to_vector(vs->GetComponents);`

Reply:
done